sdk/appconfiguration/Azure.ApplicationModel.Configuration/src/ConfigurationClient.cs

```diff
@@ -126,6 +141,11 @@ public virtual Response<ConfigurationSetting> Add(ConfigurationSetting setting,
                         throw response.CreateRequestFailedException();
                 }
             }
+            catch (Exception e)
+            {
+                scope.Failed(e);
```

Comment:
we need to think more on how to track status of Activity. Exceptions are not perfect (they are likely handled). Will we parse it (with type +something like `WebException.Status` ) in opentelemetry?

Reply:
Why do we need to parse it?

Reply:
I mean from the exception type we can understand if it is let's say timeout or HttpRequestException. 
In HttpRequestException, it is useful to know the failure details (like these https://docs.microsoft.com/en-us/dotnet/api/system.net.webexceptionstatus?view=netframework-4.8)


One thing that bothers me with exception is this:
- request starts, tracing system  doesn't care about Start
- exception happens - tracing system now have to store it and map it to what? It didn't event create Span yet. So it'll set something on the Activity tags or will be forced to create Span on start and access current span to attach exception to. 
- request ends - ok, now tracing system tracks Span and can set status


Other approaches could be:
- set status on the Activity.Tags - but it has to be a string
- set status in the Stop payload - we need to agree on the contract (prop name, value type)

Let's discuss it tomorrow. Activity kinda has to have status and if we agree that in the long term it should be there - then I like the Exception approach now.